pyasdf/schema.py

```diff
@@ -198,62 +201,62 @@ def validate_remove_default(validator, properties, instance, schema):
 REMOVE_DEFAULTS['properties'] = validate_remove_default
 
 
+
 @lru_cache()
 def _create_validator(validators=YAML_VALIDATORS):
-    validator = mvalidators.create(
-        meta_schema=load_schema(
-            'http://stsci.edu/schemas/yaml-schema/draft-01',
-            mresolver.default_url_mapping),
-        validators=validators)
-
-    validator.orig_iter_errors = validator.iter_errors
-
-    # We can't validate anything that looks like an external
-    # reference, since we don't have the actual content, so we
-    # just have to defer it for now.  If the user cares about
-    # complete validation, they can call
-    # `AsdfFile.resolve_references`.
-    def iter_errors(self, instance, _schema=None, _seen=set()):
-        if id(instance) in _seen:
-            return
-
-        if _schema is None:
-            schema = self.schema
-        else:
-            schema = _schema
-
-        if ((isinstance(instance, dict) and '$ref' in instance) or
-            isinstance(instance, reference.Reference)):
-            return
-
-        if _schema is None:
-            tag = getattr(instance, '_tag', None)
-            if tag is not None:
-                schema_path = self.ctx.tag_to_schema_resolver(tag)
-                if schema_path != tag:
-                    s = load_schema(schema_path, self.ctx.url_mapping)
-                    if s:
-                        with self.resolver.in_scope(schema_path):
-                            for x in self.orig_iter_errors(instance, s):
-                                yield x
-
-            if isinstance(instance, dict):
-                new_seen = _seen | set([id(instance)])
-                for val in six.itervalues(instance):
-                    for x in self.iter_errors(val, _seen=new_seen):
-                        yield x
-
-            elif isinstance(instance, list):
-                new_seen = _seen | set([id(instance)])
-                for val in instance:
-                    for x in self.iter_errors(val, _seen=new_seen):
-                        yield x
-        else:
-            for x in self.orig_iter_errors(instance, _schema=schema):
-                yield x
-
-    validator.iter_errors = iter_errors
-    return validator
+    meta_schema = load_schema(YAML_SCHEMA_METASCHEMA_ID,
+                              mresolver.default_url_mapping)
+    base_cls = mvalidators.create(meta_schema=meta_schema,
+                                  validators=validators)
+
+    class ASDFValidator(base_cls):
+        DEFAULT_TYPES = base_cls.DEFAULT_TYPES
```

Comment:
We should copy the dictionary here, otherwise this has the side effect of accepting tuples for *all* users of jsonschema in the process.